index.js

```diff
@@ -66,7 +68,7 @@ var S3Sync = klass(function (config, options) {
         this.writeDigestFile(function (err, resp) {
           debug('wrote digest file')
           this.options.complete && this.options.complete()
-        })
+        }.bind(this))
```

Comment:
Without this bind, the `complete` property wasn't getting called and the process hung around. Also the lock wasnt released. Was this the brokenness in those versions?